test/jdk/java/awt/TrayIcon/ShowAfterDisposeTest.java

```diff
@@ -0,0 +1,109 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 6384984 8004032
+ * @library ../regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException
+ * @summary TrayIcon try to dispay a tooltip when is not visible
+ * @run main/manual ShowAfterDisposeTest
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.SystemTray;
+import java.awt.TrayIcon;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+public class ShowAfterDisposeTest {
+    JFrame frame;
+
+    public static void main(String[] args) throws Exception {
+        boolean traySupported = SystemTray.isSupported();
+        if (!traySupported) {
+            throw new jtreg.SkippedException("The test cannot be run because SystemTray is not supported.");
+        }
+
+        ShowAfterDisposeTest test = new ShowAfterDisposeTest();
+        test.startTest();
+    }
+
+    public void startTest() throws Exception {
+        String instructions =
+            "1) When the test starts an icon is added to the SystemTray area.\n" +
+            "2a) If you use Apple OS X,\n" +
+            "    right click on this icon (it's important to click before the tooltip is shown).\n" +
+            "    The icon should disappear.\n" +
+            "2b) If you use other os (Windows, Linux, Solaris),\n" +
+            "    double click on this icon (it's important to click before the tooltip is shown).\n" +
+            "    The icon should disappear.\n" +
+            "3) If the bug is reproducible then the test will fail without assistance.\n" +
+            "4) Just press the 'pass' button.";
+
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("Test Instructions Frame")
+                .instructions(instructions)
+                .testTimeOut(10)
+                .rows(10)
+                .columns(45)
+                .build();
+        showFrameAndIcon();
+        passFailJFrame.awaitAndCheck();
```

Comment:
May I suggest simplifying it even further?

```suggestion
        PassFailJFrame.builder()
                      .title("Test Instructions Frame")
                      .instructions(instructions)
                      .testTimeOut(10)
                      .rows(10)
                      .columns(45)
                      .testUI(ShowAfterDisposeTest::showFrameAndIcon)
                      .build()
                      .awaitAndCheck();
```

where `showFrameAndIcon` is modified like this:

```java
    private static JFrame showFrameAndIcon() {
        JFrame frame = new JFrame("ShowAfterDisposeTest");
        frame.setLayout(new BorderLayout());

        // Rest of the code and no `invokeAndWait`
        // You'll have to add a try block around `tray.add(icon)` to catch and
        // wrap AWTException into RuntimeException.

        return frame;
    }

Essentially, the additional frame is not required at all, it can be dropped but I haven't explored it.